src/SchedulerState.java

```diff
@@ -475,7 +503,7 @@ private void checkReachedTargetFloor() {
         ElevatorStatus elevatorStatus = schedulerContext.getElevatorStatus();
         ElevatorTaskQueue taskQueue = schedulerContext.getElevatorTaskQueueHashMap().get(elevatorStatus.getElevatorId());
         // Don't need to process the message if the elevator has not changed floors
-        if ((schedulerContext.getPrevElevatorStatus() != null) && (elevatorStatus.getCurrentFloor() != schedulerContext.getPrevElevatorStatus().getCurrentFloor())) {
+        if ((taskQueue.getPrevElevatorStatus() != null) && (elevatorStatus.getCurrentFloor() != taskQueue.getPrevElevatorStatus().getCurrentFloor())) {
```

Comment:
This if statement may need to be modified for better logic to determine if an elevator status should be processed.

Reply:
Scheduler only processes elevator status if it mentions that the doors are open